test/synth-sm/04_loop_sat.c

```diff
@@ -0,0 +1,46 @@
+// RUN: %sea smt %s --step=small -o %t.smt2
+// RUN: %z3 %t.smt2 fp.spacer.order_children=2 2>&1 | OutputCheck %s
+// CHECK: ^sat$
+
+#include "seahorn/seahorn.h"
+
+extern int nd1();
+extern int nd2();
+extern int nd3();
+extern int nd4();
+
+// Loop invariant.
+extern int infer(int x, int n);
+int __attribute__((annotate("partial")))
+    __attribute__((noinline))
+    inv(int x, int n)
```

Comment:
are synthesized functions restricted to scalar arguments? Can a pointer argument be passed? What would that mean? Do we need this for our use case?

If we allow for pointer arguments, we need to pass implicit memory regions (arrays) that these pointers index into. This is not hard conceptually, but might be harder due to interplay with alias analysis. 

Synthesized functions are read-only, this might simplify things...

Reply:
I was working under the assumption that the synthesized functions were read-only. We do not need to pass pointers for our use-case.

Is `read-only` an attribute in llvm? I recall seeing that in the source code, but I can't find documentation for it.

Reply:
you can mark function in LLVM as only reading memory. This affects optimizations. I was wondering less about marking and optimizations, and more about the behavior of the function. If there are no pointers for now, we don't need to get into this discussion.